libbeat/cmd/instance/ilm.go

```diff
@@ -0,0 +1,168 @@
+	"fmt"
+
+	"github.com/pkg/errors"
+
+	"github.com/elastic/beats/libbeat/common"
+	"github.com/elastic/beats/libbeat/logp"
+	"github.com/elastic/beats/libbeat/outputs/elasticsearch"
+)
+
+type ilmConfig struct {
+	WriteAlias string `config:"write_alias" `
+	Name       string `config:"index.name"`
+	Pattern    string `config:"index.pattern"`
+}
+
+// Build and return a callback to load index template into ES
+func (b *Beat) writeAliasLoadingCallback() (func(esClient *elasticsearch.Client) error, error) {
+	callback := func(esClient *elasticsearch.Client) error {
+		// TODO: Need to detect ES version as older version will reject the template
+		if b.Config.ILM == nil {
+			b.Config.ILM = common.NewConfig()
+		}
+
+		config, err := getILMConfig(b)
+		if err != nil {
+			return err
+		}
+
+		err = checkElasticsearchVersionIlm(esClient)
+		if err != nil {
+			return err
+		}
+
+		firstIndex := fmt.Sprintf("%s-%s-%s", config.Name, b.Info.Version, config.Pattern)
+		// Create write alias
+		status, b, err := esClient.Request("HEAD", "/_alias/"+config.WriteAlias, "", nil, nil)
+		if err != nil && status != 404 {
+			logp.Err("Failed create write alias: %s: %+v", err, string(b))
+			return errors.Wrap(err, "failed to check for write alias")
+		}
+		if status == 200 {
+			logp.Info("Write alias already exists")
+			return nil
+		}
+
+		body := common.MapStr{
+			"aliases": common.MapStr{
+				config.WriteAlias: common.MapStr{
+					"is_write_index": true,
+				},
+			},
+		}
+
+		// Create write alias
+		_, res, err := esClient.Request("PUT", "/"+firstIndex, "", nil, body)
+		if err != nil {
+			logp.Err("Error alias creating stuff: %s, %s", err, string(res))
+			return errors.Wrap(err, "failed to create write alias: "+string(res))
+		}
+
+		logp.Info("Write alias created")
+
+		return nil
+	}
+
+	return callback, nil
+}
+
+func (b *Beat) loadILMPolicy() error {
+
+	// TODO: Implement command to load policy, get it from file?
+	outCfg := b.Config.Output
+
+	if outCfg.Name() != "elasticsearch" {
+		return fmt.Errorf("Policy loading requested but the Elasticsearch output is not configured/enabled")
+	}
+
+	esConfig := outCfg.Config()
+
+	esClient, err := elasticsearch.NewConnectedClient(esConfig)
+	if err != nil {
+		return err
+	}
+
+	// TODO: Hardcode it for now or load from json file? Should any of this be configurable in the Beat setup config?
+	policy := common.MapStr{
+		"policy": common.MapStr{
+			"type": "timeseries",
+			"phases": common.MapStr{
+				"hot": common.MapStr{
+					"actions": common.MapStr{
+						"rollover": common.MapStr{
+							"max_size": "25gb",
+						},
+					},
+				},
+			},
+		},
+	}
+	_, _, err = esClient.Request("PUT", "/_ilm/beats-default-policy", "", nil, policy)
+	return err
+}
+
+func loadConfigWithDefaults(config *ilmConfig, b *Beat) {
+	if config.WriteAlias == "" {
+		config.WriteAlias = fmt.Sprintf("%s-%s", b.Info.Beat, b.Info.Version)
+	}
+
+	if config.Pattern == "" {
+		// TODO: how many 0 should we have here?
```

Comment:
@colings86 Any opinions here?

Reply:
the rollover API will have four 0's (so `index-00003`), its not going to break if you have a different padding as it resolves the numeric value. However it would be good to be consistent here so the indexes all look the same when listed

Reply:
Will use 5 digits, 4 zero's.